SQL-Solutions.txt

```diff
@@ -0,0 +1,2743 @@
+            snum
+        FROM
+            orders);
+
+Output - 
+
++------+---------+-----------+------+
+| SNUM | SNAME   | CITY      | COMM |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1002 | Serres  | SanJose   |   13 |
+| 1003 | AxelRod | New York  |   10 |
+| 1004 | Motika  | London    |   11 |
+| 1007 | Rifkin  | Barcelona |   15 |
++------+---------+-----------+------+
+
+-- 6) List names of all customers matched with the salespeople serving them. 
+
+Query -
+
+SELECT 
+    c.cname AS CUSTOMER_NAME, s.sname AS SERVING_BY
+FROM
+    customers AS c,
+    salespeople AS s
+WHERE
+    s.snum = c.snum;
+
+Output -
+
++---------------+------------+
+| CUSTOMER_NAME | SERVING_BY |
++---------------+------------+
+| Hoffman       | Peel       |
+| Giovanni      | AxelRod    |
+| Liu           | Serres     |
+| Grass         | Serres     |
+| Clemens       | Peel       |
+| Pereira       | Motika     |
+| Cisneros      | Rifkin     |
++---------------+------------+
+
+-- 7) Find the names and numbers of all salespeople who have more than one customer.
+
+Query - 
+
+SELECT 
+    s.sname, s.snum
+FROM
+    salespeople AS s,
+    customers AS c
+WHERE
+    s.snum = c.snum
+GROUP BY c.snum
+HAVING COUNT(c.snum) > 1;
+
+Output -
+
++--------+------+
+| sname  | snum |
++--------+------+
+| Peel   | 1001 |
+| Serres | 1002 |
++--------+------+
+
+-- 8) Count the orders of each of the salespeople and output the results in descending 
+-- order. 
+
+Query -
+
+SELECT 
+    s.sname, s.snum, COUNT(o.snum) AS order_count
+FROM
+    salespeople AS s,
+    orders AS o
+WHERE
+    s.snum = o.snum
+GROUP BY o.snum
+ORDER BY order_count DESC;
+
+Output - 
+
++---------+------+-------------+
+| sname   | snum | order_count |
++---------+------+-------------+
+| Peel    | 1001 |           3 |
+| Serres  | 1002 |           3 |
+| Rifkin  | 1007 |           2 |
+| AxelRod | 1003 |           1 |
+| Motika  | 1004 |           1 |
++---------+------+-------------+
+
+-- 9) List the customer table if and only if one or more of the customers in the Customer table are located in SanJose. 
+
+Query - 
+
+SELECT 
+    *
+FROM
+    customers
+WHERE
+    (SELECT 
+            COUNT(city)
+        FROM
+            customers
+        WHERE
+            city = 'SanJose') >= 1;
+
+Output -
+
++------+----------+---------+--------+------+
+| CNUM | CNAME    | CITY    | RATING | SNUM |
++------+----------+---------+--------+------+
+| 2001 | Hoffman  | London  |    100 | 1001 |
+| 2002 | Giovanni | Rome    |    200 | 1003 |
+| 2003 | Liu      | SanJose |    200 | 1002 |
+| 2004 | Grass    | Berlin  |    300 | 1002 |
+| 2006 | Clemens  | London  |    100 | 1001 |
+| 2007 | Pereira  | Rome    |    100 | 1004 |
+| 2008 | Cisneros | SanJose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+-- 10) Match salespeople to customers according to what city they live in. 
+
+Query - 
+
+SELECT 
+    s.sname, c.cname, s.city
+FROM
```

Comment:
Have you not learned any another type of Join ?

Reply:
Used it for simpler syntaxt.  Changed it